Source/AssetRipper.SourceGenerated.Extensions/SerializedProgramExtensions.cs

```diff
@@ -7,18 +9,56 @@ public static class SerializedProgramExtensions
 		public static int GetTierCount(this ISerializedProgram program)
 		{
 			int tierCount = 1;
-			int tier = program.SubPrograms[0].ShaderHardwareTier;
-			for (int i = 1; i < program.SubPrograms.Count; i++)
+			if (program.Has_PlayerSubPrograms())
 			{
-				if (program.SubPrograms[i].ShaderHardwareTier <= tier)
+				// where is tier in PlayerSubProgram?
+			}
+			else
+			{
+				int tier = program.SubPrograms[0].ShaderHardwareTier;
+				for (int i = 1; i < program.SubPrograms.Count; i++)
 				{
-					break;
-				}
+					if (program.SubPrograms[i].ShaderHardwareTier <= tier)
+					{
+						break;
+					}
 
-				tierCount++;
+					tierCount++;
+				}
 			}
 
 			return tierCount;
 		}
+
+		public static AssetList<SerializedPlayerSubProgram> GetPlayerSubPrograms(this ISerializedProgram program)
+		{
+			if (program.Has_PlayerSubPrograms())
+			{
+				for (int i = 0; i < program.PlayerSubPrograms.Count; i++)
+				{
+					if (program.PlayerSubPrograms[i].Count > 0)
+					{
+						return program.PlayerSubPrograms[i];
+					}
+				}
+			}
+			return new AssetList<SerializedPlayerSubProgram>();
```

Comment:
Change the return type to `IReadOnlyList<SerializedPlayerSubProgram>` and this line to `return [];`

Reply:
Actually, throwing an exception here might be better if the result should never have a count of 0. Otherwise, we'll just get an index out of range exception later.